compiler/src/intermediate_representation/call_bucket.rs

```diff
@@ -107,7 +109,7 @@ impl WriteLLVMIR for CallBucket {
         Self::manage_debug_loc_from_curr(producer, self);
 
         // Check arena_size==0 which indicates arguments should not be placed into arena
-        let arena_size = self.arena_size;
+        let arena_size = u32::try_from(self.arena_size).expect("overflow");
```

Comment:
Why do we need to cast to 32bits here?

Reply:
I'll move it later. The array_type function needs it.